### PR TITLE
Added ws2811 via SPI service, default for RaspberryPi

### PR DESCRIPTION
**Pull request by zdzarsky**
Did a simple test on Pi, it's working

### CODE REVIEW

**Comment by Maheliusz** [comment]
Please, check if it's working for enough colors

File: mrc/externals/diode.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        if len(rgb) != 3:
+            raise ValueError("Length of color must be 3, found {}".format(len(rgb)))
+        for color in rgb:
+            if color not in range(0, 256):
+                raise ValueError("Diode colors must be an iterable structure [r,g,b] for rgb in ranges 0,255")
+        return True
+
+    def send_color_via_spi(self):
+        wiringpi.wiringPiSetupGpio()
+        wiringpi.wiringPiSPISetup(self.spi_ch, self.spi_speed)
```
Comment (Lukasz1928) [inline mrc/externals/diode.py]: Shouldn't SPI parameters be set once? We're waisting plenty of time!

File: mrc/externals/diode.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            'b': rgb[2]
+        }
+        self.spi_ch = spi_ch
+        self.spi_speed = spi_speed
+
+    def __call__(self):
+        self.send_color_via_spi()
+
+    def __setitem__(self, key, value):
+        if type(value) is int and type in range(0, 256) and key in ['r,g,b']:
```
Comment (Lukasz1928) [inline mrc/externals/diode.py]: I think it should be "value" instead of "type" in 
if ... and type in range(0, 256) ...

Also, range(0, 256) = range(256)

Comment (zdzarsky) [reply mrc/externals/diode.py]: Geez.

**Pull request merged**